tests/test_CompareEclipseRestart.cpp

```diff
@@ -0,0 +1,173 @@
+    }
+}
+
+void ReadUNRST::close() {
+    ecl_file_close(ecl_file1);
+    ecl_file_close(ecl_file2);
+}
+
+bool ReadUNRST::results(const char* keyword, std::vector<double>* absDeviation, std::vector<double>* relDeviation) {
+    if (!ecl_file_has_kw(ecl_file1, keyword) || !ecl_file_has_kw(ecl_file2, keyword)) {
+        std::cout << "The file does not have this keyword." << std::endl;
+        return false;
+    }
+    unsigned int occurrences1 = ecl_file_get_num_named_kw(ecl_file1, keyword);
+    unsigned int occurrences2 = ecl_file_get_num_named_kw(ecl_file2, keyword);
+    if (occurrences1 != occurrences2) {
+        std::cout << "Keyword occurrences are not equal." << std::endl;
+        return false;
+    }
+
+    for (unsigned int index = 0; index < occurrences1; ++index) {
+        ecl_kw_type* ecl_kw1 = ecl_file_iget_named_kw(ecl_file1, keyword, index);
+        unsigned int numActiveCells1 = ecl_kw_get_size(ecl_kw1);
+        ecl_kw_type* ecl_kw2 = ecl_file_iget_named_kw(ecl_file2, keyword, index);
+        unsigned int numActiveCells2 = ecl_kw_get_size(ecl_kw2);
+        if (numActiveCells1 != numActiveCells2) {
+            std::cout << "Number of active cells are different." << std::endl;
+            return false;
+        }
+
+        std::vector<double> values1; // Elements in the vector corresponds to active cells
+        std::vector<double> values2; // Elements in the vector corresponds to active cells
+        values1.resize(numActiveCells1);
+        values2.resize(numActiveCells2);
+        ecl_kw_get_data_as_double(ecl_kw1, values1.data());
+        ecl_kw_get_data_as_double(ecl_kw2, values2.data());
+
+        for (unsigned int i = 0; i < values1.size(); i++) {
+            calculateDeviations(absDeviation, relDeviation, values1[i], values2[i]);
+        }
+    }
+    return true;
+}
+
+
+void ReadUNRST::calculateDeviations(std::vector<double> *absDeviationVector,std::vector<double> *relDeviationVector, double val1, double val2){
+    double absDeviation = 0;
+    double relDeviation = 0;
+    if (val1 < 0) {
+        val1 = 0;
+        //std::cout << "Value 1 has a negative quantity." << std::endl; 
+    }
+    if (val2 < 0) {
+        val2 = 0;
+        //std::cout << "Value 2 has a negative quantity." << std::endl; 
+    }
+
+    if (!(val1 == 0 && val2 == 0)) {
+        absDeviation = std::max(val1, val2) - std::min(val1, val2);
+        absDeviationVector->push_back(absDeviation);
+        if (val1 != 0 && val2 != 0) {
+            relDeviation = absDeviation/static_cast<double>(std::max(val1, val2));
+            relDeviationVector->push_back(relDeviation);
+        }
+    }
+}
+
+double ReadUNRST::max(std::vector<double>& vec){
+    double maxValue = 0;
+    for(unsigned int i = 0; i < vec.size(); ++i){
+        if (vec[i] > maxValue) {
+            maxValue = vec[i];
+        }
+    }
+    return maxValue;
+}
+
+double ReadUNRST::median(std::vector<double>& vec) {
+    if(vec.empty()) return 0;
```

Comment:
Always use braces for if/else/for/while bodies, I do not merge otherwise (except by accident...)

Also, here you should use nth_element instead of sort, see http://stackoverflow.com/questions/1719070/what-is-the-right-approach-when-using-stl-container-for-median-calculation/1719155#1719155